### PR TITLE
Fix a conflict case for io.cozy.sharings doc

### PR DESCRIPTION
**Pull request by nono**
When a sharing is revoked, it may happen that a conflict (409 from CouchDB) is received when updating the io.cozy.sharings document after deleting its triggers. After that, an error trigger_on_inactive_sharing is reported by cozy-stack check sharings.

And the fix is quite simple: when we receive a 409, we reload the document from CouchDB and try again.

### CODE REVIEW

File: model/sharing/sharing.go [hunk 1]
```diff
@@ -553,17 +553,31 @@ func (s *Sharing) RevokeByNotification(inst *instance.Instance) error {
...
-	for i, m := range s.Members {
-		if i > 0 && m.Instance != "" {
-			s.Members[i].Status = MemberStatusRevoked
+	var err error
+	for i := 0; i < 3; i++ {
+		s.Triggers = Triggers{}
+		s.Credentials = nil
+		s.Active = false
+
+		for i, m := range s.Members {
```
Comment (taratatach) [inline model/sharing/sharing.go:563]: No issues with reusing the same variable `i` here?

File: model/sharing/sharing.go [hunk 2]
```diff
@@ -553,17 +553,31 @@ func (s *Sharing) RevokeByNotification(inst *instance.Instance) error {
...
+			if i > 0 && m.Instance != "" {
```
Comment (taratatach) [inline model/sharing/sharing.go:564]: The idea is to not revoke the owner of the sharing?

File: model/sharing/sharing.go [hunk 3]
```diff
@@ -553,17 +553,31 @@ func (s *Sharing) RevokeByNotification(inst *instance.Instance) error {
...
 	}
-	s.Credentials = nil
-	s.Active = false
 
-	for i, m := range s.Members {
-		if i > 0 && m.Instance != "" {
-			s.Members[i].Status = MemberStatusRevoked
+	var err error
+	for i := 0; i < 3; i++ {
+		s.Triggers = Triggers{}
```
Comment (taratatach) [inline model/sharing/sharing.go:559]: What's the idea here? Haven't we removed all the triggers already at that point?

**Comment by taratatach** [comment]
Out of curiosity, why do we have a `trigger_on_inactive_sharing` error when we have a sharing with no triggers? I would have expected something like `sharing_without_triggers`.

**Comment by nono** [comment]
We have the triggers, but also a reference to them in the `io.cozy.sharings` document. When the conflict blocks the `io.cozy.sharings` doc update, the reference is still here. And the stack just look at the reference for the `trigger_on_inactive_sharing` error (not that the triggers really exist or not).

File: model/sharing/sharing.go (hunk 1 above)
Comment (nono) [reply model/sharing/sharing.go:563]: No, we are using `:=` which declares a new variable (and mask the previous one for this range).

File: model/sharing/sharing.go (hunk 2 above)
Comment (nono) [reply model/sharing/sharing.go:564]: Yes, it's the idea.

**Review by taratatach — APPROVED** [review]

**Pull request merged**